helper/common/common.go

```diff
@@ -174,3 +175,48 @@ func PadLeftOrTrim(bb []byte, size int) []byte {
 
 	return tmp
 }
+
+// FetchWhitelistFromConfig fetches whitelist object from the config
+// if doesn't exist returns empty map
+func FetchWhitelistFromConfig(genesisConfig *chain.Chain) map[string]interface{} {
+	// Fetch whitelist if exists, if not init
+	whitelistConfig := genesisConfig.Params.Whitelists
+	if len(whitelistConfig) == 0 {
+		whitelistConfig = make(map[string]interface{})
+	}
+
+	return whitelistConfig
+}
+
+// FetchDeploymentWhitelist fetches deployment whitelist from the genesis config
+// if doesn't exist returns empty list
+func FetchDeploymentWhitelist(genesisConfig *chain.Chain) ([]types.Address, error) {
+	// Fetch whitelist config if exists, if not init
+	whitelistConfig := FetchWhitelistFromConfig(genesisConfig)
+
+	// Extract deployment whitelist if exists, if not init
+
+	var deploymentWhitelistRaw []interface{}
+
+	if whitelistConfig["deployment"] != nil {
+		var ok bool
+
+		deploymentWhitelistRaw, ok = whitelistConfig["deployment"].([]interface{})
+		if !ok {
+			return nil, errors.New("invalid type assertion for deployment whitelist")
```

Comment:
This should be defined as a local constant error

Reply:
Fixed 79615b617501fca361343f47684ab6a5b9127473